### PR TITLE
feat(extract): adds recognition of jsdoc @import type imports

### PR DESCRIPTION
**Pull request by sverweij**
## Description

- detects [jsdoc @import imports](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#the-jsdoc-import-tag) added in TS5.5
- adds a new `jsdoc`, `jsdoc-import-tag` dependency types
- adds unit tests
- dog foods the `@import` jsdoc tag
- [x] feature switches it in the configuration (otherwise it would be a breaking change) - 
- [x] feature switch true implies `parser` set to `tsc`
- [x] document it (and add caveats (TS 5.5 and up - `parser` attribute set to `tsc`)
- [x] adds e2e test
- 🏕️ disables the yarn berry integration test - for #reasons it now acts up on the ci (node 23 issue? Something else?), where it runs perfectly fine on local machine - something to figure out later.

Future work:
- also detect jsdoc-bracket-import (`/** @type {import('yadda').bla} */`) statements in other jsdoc tags - see [typescript jsdoc reference](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) for a list => likely better in a _separate PR_.

## Motivation and Context

Addresses #964 

## How Has This Been Tested?

- [x] green ci
- [x] additional automated non-regression tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation only change
- [ ] Refactor (non-breaking change which fixes an issue without changing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] :book:

  - My change doesn't require a documentation update, or ...
  - it _does_ and I have updated it

- [x] :balance_scale:
  - The contribution will be subject to [The MIT license](https://github.com/sverweij/dependency-cruiser/blob/main/LICENSE), and I'm OK with that.
  - The contribution is my own original work.
  - I am ok with the stuff in [**CONTRIBUTING.md**](https://github.com/sverweij/dependency-cruiser/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by louwers** [comment]
Test driving it, seems to work great already.

**Comment by sverweij** [comment]
> Test driving it, seems to work great already.

Thanks! Good to hear it works 🎉 

There will be small changes over time. E.g. since yesterday I've changed the names of dependencyTypes a bit (for regular cli use shouldn't matter) and I'll add a feature switch to (1) prevent this being a breaking change (2) auto-switch the javascript parser to `tsc`, which is the only one who can detect these.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ env:
   CI: true
   # the LANG (used by the Intl API, which we use for some of the number
   # formatting) is set in the run scripts. This works everywhere, except on
-  # windows. So, especially for windows ...
+  # windows. So, especinodeally for windows ...
```
Comment (sverweij) [inline .github/workflows/ci.yml:16]:
```suggestion
  # windows. So, especially for windows ...
```

File: doc/options-reference.md [hunk 2]
```diff
@@ -784,6 +785,33 @@ you can provide the parameters like so:
...
+
+As currently on the TypeScript compiler (`tsc`) can detect these imports, switching
+on this option implies dependency-cruiser will set `options.parser` to `tsc` so
+it uses the TypeScript compiler to parse not only TypeScript but also JavaScript.
+
+```json
+"options": {
+  "detectJSDocImports": true // implies `parser: "tsc"`
+}
+```
```
Comment (sverweij) [inline doc/options-reference.md:806]:
```suggestion
```javascript
options: {
  detectJSDocImports: true // implies `parser: "tsc"`
}
```
```

**Pull request merged**